### PR TITLE
jit: support for store instructions

### PR DESCRIPTION
**Pull request by mathetake**

### CODE REVIEW

File: wasm/jit/jit_amd64_test.go [hunk 1]
```diff
@@ -1684,6 +1684,213 @@ func TestAmd64Builder_handleLoad32(t *testing.T) {
...
+
+			// All the values are popped, so the stack pointer must be zero.
+			require.Equal(t, uint64(0), eng.stackPointer)
+			// Check the stored value.
+			offset := o.Arg.Offest + baseOffset
+			switch o.Type {
+			case wazeroir.UnsignedTypeI32, wazeroir.UnsignedTypeF32:
+				v := binary.LittleEndian.Uint32(mem.Buffer[offset : offset+4])
+				require.Equal(t, uint32(storeTargetValue), v)
+				// The trailing bytes must be intact since this is 32-bit mov.
```
Comment (codefromthecrypt) [inline wasm/jit/jit_amd64_test.go:1745]: 👍

File: wasm/jit/jit_amd64_test.go [hunk 2]
```diff
@@ -1684,6 +1684,213 @@ func TestAmd64Builder_handleLoad32(t *testing.T) {
...
+		uintptr(unsafe.Pointer(eng)),
+		uintptr(unsafe.Pointer(&mem.Buffer[0])),
+	)
+
+	// All the values are popped, so the stack pointer must be zero.
+	require.Equal(t, uint64(0), eng.stackPointer)
+	// Check the stored value.
+	offset := o.Arg.Offest + baseOffset
+	require.Equal(t, uint32(storeTargetValue), binary.LittleEndian.Uint32(mem.Buffer[offset:]))
+	// The traiing bytes must be intact since this is only moving 4 byte.
```
Comment (codefromthecrypt) [inline wasm/jit/jit_amd64_test.go:1890]:
```suggestion
	// The trailing bytes must be intact since this is only moving 4 byte.
```

**Review by codefromthecrypt — APPROVED** [review]
well done!

**Comment by mathetake** [comment]
img

**Pull request merged**